[Bug]: share_plus_platform_interface 3.1.0 is incompatible with Flutter < 3.3.0

### Platform

Flutter 3.0.5

### Plugin

share_plus_platform_interface

### Version

3.1.0

### Flutter SDK

3.0.5

### Steps to reproduce

1. Install Flutter 3.0.5
2. Add `share_plus: 4.5.0`, `share_plus_platform_interface: 3.1.0` to `pubspec.yaml`
3. Execute `flutter run` or `flutter build`
4. Build fail

### Code Sample

```dart
dependencies:
  flutter:
    sdk: flutter

  share_plus: 4.5.0
  share_plus_platform_interface: 3.1.0
```


### Logs

```shell
timekone$ flutter run --verbose
[ +112 ms] executing: sysctl hw.optional.arm64
[  +26 ms] Exit code 1 from: sysctl hw.optional.arm64
[        ] sysctl: unknown oid 'hw.optional.arm64'
[   +8 ms] executing: [/Users/timekone/development/flutter/] git -c
log.showSignature=false log -n 1 --pretty=format:%H
[  +34 ms] Exit code 0 from: git -c log.showSignature=false log -n 1
--pretty=format:%H
[        ] f1875d570e39de09040c8f79aa13cc56baab8db1
[        ] executing: [/Users/timekone/development/flutter/] git tag --points-at
f1875d570e39de09040c8f79aa13cc56baab8db1
[ +167 ms] Exit code 0 from: git tag --points-at
f1875d570e39de09040c8f79aa13cc56baab8db1
[        ] 3.0.5
[   +9 ms] executing: [/Users/timekone/development/flutter/] git rev-parse
--abbrev-ref --symbolic @{u}
[  +15 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [/Users/timekone/development/flutter/] git ls-remote
--get-url origin
[  +15 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[ +106 ms] executing: [/Users/timekone/development/flutter/] git rev-parse
--abbrev-ref HEAD
[  +20 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[   +8 ms] executing: sw_vers -productName
[  +20 ms] Exit code 0 from: sw_vers -productName
[        ] macOS
[        ] executing: sw_vers -productVersion
[  +15 ms] Exit code 0 from: sw_vers -productVersion
[        ] 12.6
[        ] executing: sw_vers -buildVersion
[  +18 ms] Exit code 0 from: sw_vers -buildVersion
[        ] 21G115
[        ] executing: uname -m
[   +6 ms] Exit code 0 from: uname -m
[        ] x86_64
[  +66 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping
update.
[   +4 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required,
skipping update.
[   +1 ms] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required,
skipping update.
....
[   +7 ms] executing: /Users/timekone/Library/Application
Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7351085/Android
Studio.app/Contents/jre/jdk/Contents/Home/bin/java -version
[ +288 ms] Exit code 0 from: /Users/timekone/Library/Application
Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7351085/Android
Studio.app/Contents/jre/jdk/Contents/Home/bin/java -version
[   +1 ms] openjdk version "11.0.8" 2020-07-14
           OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
           OpenJDK 64-Bit Server VM (build 11.0.8+10-b944.6916264, mixed mode)
[   +1 ms] executing:
[/Users/timekone/AndroidStudioProjects/flutter/share_plus_test/android/]
/Users/timekone/AndroidStudioProjects/flutter/share_plus_test/android/gradlew
-Pverbose=true -Ptarget-platform=android-arm64
-Ptarget=/Users/timekone/AndroidStudioProjects/flutter/share_plus_test/lib/main.
dart -Pbase-application-name=android.app.Application
-Pdart-defines=RkxVVFRFUl9XRUJfQVVUT19ERVRFQ1Q9dHJ1ZQ== -Pdart-obfuscation=false
-Ptrack-widget-creation=true -Ptree-shake-icons=false
-Pfilesystem-scheme=org-dartlang-root assembleDebug
[+1946 ms]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:25:5: Error: Type 'Rect' not found.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[   +4 ms]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:50:5: Error: Type 'Rect' not found.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[        ]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:78:5: Error: Type 'Rect' not found.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[        ]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:107:5: Error: Type 'Rect' not
found.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[        ]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:140:5: Error: Type 'Rect' not
found.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[+3392 ms] > Task :app:preBuild UP-TO-DATE
[        ] > Task :app:preDebugBuild UP-TO-DATE
[        ] > Task :app:mergeDebugNativeDebugMetadata NO-SOURCE
[+2347 ms]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:25:5: Error: 'Rect' isn't a type.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[   +1 ms]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:50:5: Error: 'Rect' isn't a type.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[   +1 ms]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:78:5: Error: 'Rect' isn't a type.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[        ]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:107:5: Error: 'Rect' isn't a type.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[   +1 ms]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:140:5: Error: 'Rect' isn't a type.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[+1526 ms] > Task :app:compileFlutterBuildDebug
[   +2 ms]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:25:5: Error: Type 'Rect' not found.
[   +3 ms]     Rect? sharePositionOrigin,
[        ]     ^^^^
[        ]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:50:5: Error: Type 'Rect' not found.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[        ]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:78:5: Error: Type 'Rect' not found.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[        ]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:107:5: Error: Type 'Rect' not
found.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[        ]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:140:5: Error: Type 'Rect' not
found.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[+3687 ms]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:25:5: Error: 'Rect' isn't a type.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[        ]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:50:5: Error: 'Rect' isn't a type.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[        ]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:78:5: Error: 'Rect' isn't a type.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[        ]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:107:5: Error: 'Rect' isn't a type.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[        ]
../../../.pub-cache/hosted/pub.dartlang.org/share_plus_platform_interface-3.1.0/
lib/method_channel/method_channel_share.dart:140:5: Error: 'Rect' isn't a type.
[        ]     Rect? sharePositionOrigin,
[        ]     ^^^^
[+4396 ms] > Task :app:compileFlutterBuildDebug FAILED
[ +101 ms] FAILURE: Build failed with an exception.
[        ] * Where:
[        ] Script
'/Users/timekone/development/flutter/packages/flutter_tools/gradle/flutter.gradl
e' line: 1156
[        ] * What went wrong:
[        ] Execution failed for task ':app:compileFlutterBuildDebug'.
[        ] > Process 'command '/Users/timekone/development/flutter/bin/flutter''
finished with non-zero exit value 1
[        ] * Try:
[        ] > Run with --stacktrace option to get the stack trace.
[        ] > Run with --info or --debug option to get more log output.
[        ] > Run with --scan to get full insights.
[        ] * Get more help at https://help.gradle.org
[        ] BUILD FAILED in 17s
[        ] Deprecated Gradle features were used in this build, making it
incompatible with Gradle 8.0.
[        ] You can use '--warning-mode all' to show the individual deprecation
warnings and determine if they come from your own scripts or plugins.
[        ] See
https://docs.gradle.org/7.4/userguide/command_line_interface.html#sec:command_li
ne_warnings
[        ] 1 actionable task: 1 executed
[ +449 ms] Running Gradle task 'assembleDebug'... (completed in 18.5s)
[   +6 ms] Exception: Gradle task assembleDebug failed with exit code 1
[   +2 ms] "flutter run" took 23,825ms.
[   +7 ms] 
           #0      throwToolExit
(package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand
           (package:flutter_tools/src/commands/run.dart:699:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure>
           (package:flutter_tools/src/runner/flutter_command.dart:1183:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure>
           (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand
(package:args/command_runner.dart:209:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure>
           (package:flutter_tools/src/runner/flutter_command_runner.dart:281:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure>
           (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand
           (package:flutter_tools/src/runner/flutter_command_runner.dart:229:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure>
           (package:flutter_tools/runner.dart:62:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure>
           (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:94:3)
           <asynchronous suspension>
           
           
[ +159 ms] ensureAnalyticsSent: 157ms
[   +2 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1

```
```


### Flutter Doctor

```shell
flutter doctor -v
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.6 21G115 darwin-x64, locale
    en-GB)
    • Flutter version 3.0.5 at /Users/timekone/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (3 months ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version
    31.0.0-rc4)
    • Android SDK at /Users/timekone/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0-rc4
    • Java binary at: /Users/timekone/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7784292/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Users/timekone/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7784292/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] Android Studio (version 4.2)
    • Android Studio at /Users/timekone/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7351085/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] VS Code (version 1.71.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.48.0

[✓] Connected device (3 available)
    • Pixel 6 (mobile) • 21081FDF6006LZ • android-arm64  • Android 13 (API 33)
    • macOS (desktop)  • macos          • darwin-x64     • macOS 12.6 21G115
      darwin-x64
    • Chrome (web)     • chrome         • web-javascript • Google Chrome
      105.0.5195.125

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```


### Checklist before submitting a bug

- [X] I Google'd a solution and I couldn't find it
- [X] I searched on StackOverflow for a solution and I couldn't find it
- [X] I read the README.md file of the plugin
- [X] I'm using the latest version of the plugin
- [X] All dependencies are up to date with `flutter pub upgrade`
- [X] I did a `flutter clean`
- [X] I tried running the example project

--- a/packages/share_plus/share_plus_platform_interface/CHANGELOG.md
+++ b/packages/share_plus/share_plus_platform_interface/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 3.1.1
+
+- Add missing dart:ui import
+
 ## 3.1.0
 
 - Add `shareXFiles` implementations

--- a/packages/share_plus/share_plus_platform_interface/lib/method_channel/method_channel_share.dart
+++ b/packages/share_plus/share_plus_platform_interface/lib/method_channel/method_channel_share.dart
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 import 'dart:async';
+import 'dart:ui';
 
 import 'package:cross_file/cross_file.dart';
 import 'package:flutter/services.dart';

--- a/packages/share_plus/share_plus_platform_interface/lib/method_channel/method_channel_share.dart
+++ b/packages/share_plus/share_plus_platform_interface/lib/method_channel/method_channel_share.dart
@@ -3,6 +3,8 @@
 // found in the LICENSE file.
 
 import 'dart:async';
+// Keep dart:ui for retrocompatiblity with Flutter <3.3.0
+// ignore: unnecessary_import
 import 'dart:ui';
 
 import 'package:cross_file/cross_file.dart';

--- a/packages/share_plus/share_plus_platform_interface/pubspec.yaml
+++ b/packages/share_plus/share_plus_platform_interface/pubspec.yaml
@@ -1,6 +1,6 @@
 name: share_plus_platform_interface
 description: A common platform interface for the share_plus plugin.
-version: 3.1.0
+version: 3.1.1
 homepage: https://plus.fluttercommunity.dev/
 repository: https://github.com/fluttercommunity/plus_plugins/tree/main/packages/
 

